src/services/awards.ts

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+
+import _ from 'lodash';
+import { Problem } from '@acmascis/codeforces-client/build/interfaces/problem.interface';
+import { Standings } from '@acmascis/codeforces-client/build/interfaces/standings.interface';
+import { Submission } from '@acmascis/codeforces-client/build/interfaces/submission.interface';
+
+let id = 0;
+const awards: { [id: number]: { teamId: number; citation: string } } = {};
+
+export const GetAwards = async (
+    standings: Standings,
+    problems: Problem[],
+    handlesIds: { [handle: string]: number },
+    submissions: Submission[],
+) => {
+    await getFirstTwo(handlesIds, submissions);
+    await getSecondTwo(handlesIds, submissions);
+    await getFirstToSolve(problems, handlesIds, submissions);
+    await getMedals(standings, handlesIds);
```

Comment:
[NITPICK] Those are not using any `async-await` syntax, so no need to mark them as async.
Also be careful of using global variables that are being edited like this. The state is kept between invokations, so if you called this function twice it will have the old state.

I know this is not the case in this app, so feel free not to fix it, but this is just a note.

Suggested alternative

```js
// making awards a local state and passing it to the functions
let awards = {}
await getFirstTwo(handlesIds, submissions, awards);
await getSecondTwo(handlesIds, submissions, awards);
```

```js
// what I prefer more is to create a new object and combine them
// this is valid if they are adding their unique keys which I think they do.
return {
	...await getFirstTwo(handlesIds, submissions),
    	...await getSecondTwo(handlesIds, submissions),
    	...await getFirstToSolve(problems, handlesIds, submissions),
    	...await getMedals(standings, handlesIds),
}
```

But a question here, isn't awards here treated as an array? why is it an object?